### PR TITLE
Refactor Diff and Patch Pages

### PR DESCRIPTION
**Pull request by mbtools**
Switch to new page model (ref #6048)

### CODE REVIEW

**Comment by mbtools** [comment]
switch between unified and split view not working yet

File: src/ui/pages/zcl_abapgit_gui_page_diff.clas.abap [hunk 1]
```diff
@@ -1,11 +1,16 @@
 CLASS zcl_abapgit_gui_page_diff DEFINITION
   PUBLIC
-  INHERITING FROM zcl_abapgit_gui_page
-  CREATE PUBLIC.
+  INHERITING FROM zcl_abapgit_gui_component
+  CREATE PRIVATE
+  GLOBAL FRIENDS zcl_abapgit_gui_page_patch.
```
Comment (larshp) [inline src/ui/pages/zcl_abapgit_gui_page_diff.clas.abap:5]: this new FRIENDS looks suspicions?

**Review by larshp — CHANGES_REQUESTED** [review]

File: src/ui/pages/zcl_abapgit_gui_page_diff.clas.abap (hunk 1 above)
Comment (mbtools) [reply src/ui/pages/zcl_abapgit_gui_page_diff.clas.abap:5]: Yeah, the UI framework isn't made for pages inheriting from other pages. I will re-do it with one super and diff and patch as children (like code inspector + syntax check)

Comment (larshp) [reply src/ui/pages/zcl_abapgit_gui_page_diff.clas.abap:5]: does it need to be inheritance? can it be done via composition? ie. extract some logic to a separate class

**Comment by mbtools** [comment]
now `create` is clean for both classes 😃

**Review by larshp — APPROVED** [review]
I've not read through it all, but lets just try merging

**Pull request merged**